src/util/GenerateProject.ts

```diff
@@ -58,83 +60,100 @@ export async function generateProject(fileUri): Promise<void> {
+            + '-i ' + this.yamlFile + ' -g java --library microprofile -o ' + this.srcDir + '\
             --api-package ' + this.package + '.api \
             --model-package ' + this.package + '.models';
         commands.push(jarCommand);
     }
-    var exec = require('child_process').exec;
-    function runCommands(array, srcDir, packagePath, callback) {
-        var index = 0;
-        var results = [];
-        function next() {
-            if (index < array.length) {
-                exec(array[index++], function (err, stdout) {
-                    if (err) { return callback(err); }
-                    // do the next iteration
-                    results.push(stdout);
-                    next();
-                });
-            } else {
-                callback(null, results, srcDir, packagePath);
-            }
-        }
-        // start the first iteration
-        next();
-    }
 
-    runCommands(commands, this.srcDir, this.packagePath, function (err, results, srcDir, packagePath) {
-        if (err) {
-            console.error(err);
-            vscode.window.showErrorMessage("An error has occurred. Unable to generate a rest client in the specified directory.  Check the debug console for more information.");
+    if (commands && commands.length > 0) {  // there are commands to run
+        var exec = require('child_process').exec;
+        function runCommands(array, srcDir, packagePath, callback) {
+            var index = 0;
+            var results = [];
+            function next() {
+                if (index < array.length) {
+                    exec(array[index++], function (err, stdout) {
+                        if (err) { return callback(err); }
+                        // do the next iteration
+                        results.push(stdout);
+                        next();
+                    });
+                } else {
+                    callback(null, results, srcDir, packagePath);
+                }
+            }
+            // start the first iteration
+            next();
         }
-        if (results) {
-            vscode.window.showInformationMessage("Successfully generated a MicroProfile Rest Client interface template.");
-            try {
-                const path = require('path');
-                let generatedPath = path.resolve(srcDir, 'src', 'gen', 'java', packagePath);
 
-                // move models and apis folders to srcDir
-                var ncp = require('ncp').ncp;
-                ncp(generatedPath, srcDir, function (err) {
-                    if (err) {
-                        return console.error(err);
-                    }
-                    console.log('Moved ' + generatedPath + ' to ' + srcDir);
+        vscode.window.withProgress( // run the command with a "progress" dialog
+			{
+				location: vscode.ProgressLocation.Notification,
+				title: "Generating the MicroProfile Rest Client interface template..."
+			}, async () => {
```

Comment:
The spacing looks odd here, should change the tabs to spaces.

Reply:
Changed to spaces